src/Support/Artisan.php

```diff
@@ -98,11 +98,22 @@ protected function getCommand($command)
      */
     public function commands()
     {
-        return collect(IlluminateArtisan::all())->mapWithKeys(function (
-            $command
-        ) {
-            return [$command->getName() => $this->getCommand($command)];
-        });
+        $whitelistedCommands = config('artisan-tool.whitelistedCommands');
+
+        return collect(IlluminateArtisan::all())
+            ->filter(function ($command) use ($whitelistedCommands) {
+                if (null === $whitelistedCommands || empty($whitelistedCommands)) {
+                    return true;
+                }
+
+                return \in_array(\get_class($command), $whitelistedCommands, true);
```

Comment:
I don't think it's necessary to take namespacing into account for global functions. Although I've seen from my colleagues that PhpStorm really wants to add those `use`-statements though
```php
return in_array(get_class($command), $whitelistedCommands, true);
```

Reply:
@jstoone it’s a performance gain. Ref: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3048